pkg/trace/traceutil/azure.go

```diff
@@ -71,27 +74,79 @@ func getAppServicesTags(getenv func(string) string) map[string]string {
 		aasSiteType:        appService,
 		aasSubscriptionID:  subscriptionID,
 	}
+
+	// Remove the Java and .NET logic once non-universal extensions are deprecated
+	if websiteOS == "windows" {
+		if extensionVersion != "" {
+			tags[aasExtensionVersion] = extensionVersion
+		} else if val := getenv("DD_AAS_JAVA_EXTENSION_VERSION"); val != "" {
+			tags[aasExtensionVersion] = val
+		} else if val := getenv("DD_AAS_DOTNET_EXTENSION_VERSION"); val != "" {
+			tags[aasExtensionVersion] = val
+		}
+	}
+
+	return tags
 }
 
 func getEnvOrUnknown(env string, getenv func(string) string) string {
 	val := getenv(env)
-	if len(env) == 0 {
+	if len(val) == 0 {
 		val = unknown
 	}
 	return val
 }
 
-func getRuntime(getenv func(string) string) (rt string) {
-	env := getenv(ddRuntime)
-	switch env {
-	case dotnetRuntime:
-		rt = dotnetFramework
-	case nodeRuntime:
-		rt = nodeFramework
+func getRuntime(websiteOS string, getenv func(string) string) (rt string) {
+	switch websiteOS {
+	case "windows":
+		rt = getWindowsRuntime(getenv)
+	case "linux", "darwin":
+		rt = getLinuxRuntime(getenv)
 	default:
 		rt = unknown
 	}
-	return
+
+	return rt
+}
+
+func getWindowsRuntime(getenv func(string) string) (rt string) {
+	if getenv("WEBSITE_STACK") == "JAVA" {
+		rt = javaFramework
+	} else if val := getenv("WEBSITE_NODE_DEFAULT_VERSION"); val != "" {
+		rt = nodeFramework
+	} else {
+		// FIXME: Windows AAS only supports Java, Node, and .NET so we can infer this
```

Comment:
It's intended for a future fix/change in case the current assumptions change

The assumption is just that there are only three runtimes (Node, Dotnet, Java). This should hold true for a while since Python and PHP were recently removed in late 2022